ui/src/app/pages/app/component.ts

```diff
@@ -66,11 +68,18 @@ export class AppComponent implements OnInit {
         console.log('To override the API URLs, try:\n' +
           'setAllOfUsApiUrl(\'https://host.example.com:1234\')');
 
-        const lastActive = window.localStorage.getItem(INACTIVITY_CONFIG.LOCAL_STORAGE_KEY_LAST_ACTIVE);
-        if (lastActive == null || Date.now() - parseInt(lastActive) > environment.inactivityTimeoutInSeconds * 1000) {
-          localStorage.setItem(INACTIVITY_CONFIG.LOCAL_STORAGE_KEY_LAST_ACTIVE, Date.now().toString());
-          window.location.assign('https://accounts.google.com/logout');
-        }
+        this.signInService.isSignedIn$.subscribe(signedIn => {
+          if (signedIn) {
+            const lastActive = window.localStorage
+              .getItem(INACTIVITY_CONFIG.LOCAL_STORAGE_KEY_LAST_ACTIVE);
+            if (lastActive == null ||
+              Date.now() - parseInt(lastActive, 10) > environment.inactivityTimeoutSeconds * 1000) {
+              localStorage
+                .setItem(INACTIVITY_CONFIG.LOCAL_STORAGE_KEY_LAST_ACTIVE, Date.now().toString());
+              SignedInComponent.prototype.navigateSignOut();
```

Comment:
This is pretty sketchy and the dependency is questionable, would just move this into the navigation utils as a static helper function